.github/workflows/ci-nix.yml

```diff
@@ -371,25 +371,33 @@ jobs:
           # Get all unique container names from the matrix
           containers=$(echo '${{ needs.containers.outputs.matrix }}' | jq -r '.[].container.name' | sort -u)
 
+          # Reference the following blog post for multi arch images with nix and docker manifest
+          # https://tech.aufomm.com/how-to-build-multi-arch-docker-image-on-nixos/?utm_source=chatgpt.com#Tag-multi-arch
           for container in $containers; do
             image="fedimint/$container"
             if [ "$GITHUB_REF" = "refs/heads/master" ]; then
               echo "Creating manifest for $image:master"
               docker manifest create "$image:master" \
                 "$image:master-x86_64-linux" \
                 "$image:master-aarch64-linux"
+              docker manifest annotate "$image:master" "$image:master-x86_64-linux" --arch amd64
+              docker manifest annotate "$image:master" "$image:master-aarch64-linux" --arch arm64
```

Comment:
Pretty sure this is racy btw, didn't consider this when PRing my initial version :disappointed:  

The "upload/download artifacts to transfer tarballs between CI jobs" approach seems preferable, but let's give this version a try.

@lukechilds pointed me to how he did it for `bitcoind`: https://github.com/lukechilds/docker-bitcoind/blob/6ebb1895a75dd4a0baaf41d5cff1b2ba81abb68a/.github/workflows/ci.yml#L60-L64